cli-commands/forgae-deploy/forgae-deployer.js

```diff
@@ -64,15 +118,29 @@ class Deployer {
         }
         const compiledContract = await client.contractCompile(contract, {
             gas
-        })
-        const deployPromise = await compiledContract.deploy(deployOptions)
+        });
+
+        const deployPromise = await compiledContract.deploy(deployOptions);
         const deployedContract = await deployPromise;
 
         let regex = new RegExp(/[\w]+.aes$/);
-        let contractFileName = regex.exec(contractPath)
+        let contractFileName = regex.exec(contractPath);
+
+        let txInfo = await getTxInfo(deployedContract.transaction, this.network);
+        const isSuccess = txInfo.gasPrice >= 0 && txInfo.gasUsed > 0;
```

Comment:
So the transaction wont use gas if the transaction has failed? I do not feel very confident in this way of figuring out if the transaction is successful